### DISCUSSION TITLE
Msi recognized as threat by Windows Defender

### DISCUSSION THREAD

**Discussion by vilu988** [oleg-shilo/wixsharp#1496] [Q&A]
I noticed that the .msi file built using wixsharp has been recognized as a 'Trojan:Script/Wacatac.H!ml' by Windows Defender for two days now. I build the installer using github actions, then download it and put it on One Drive, people who try to download the installer get a message that the file is infected with a virus. I use libraries in version 1.19.0, I have never signed the .msi file and .dll files that are part of the installation package. Has anyone encountered a similar problem or have an idea what could be the reason?

**Comment by oleg-shilo** [comment]
The reason for this is over-aggressive Windows Defender (WD) detection. Usually based on some AI algorithm. Oh my....

This has been a problem for some other products too. I remember at some point Malwarebytes was marking as infected a freshly build exe from the empty .NET console app project. 

I have no possible workaround for that. WD can declare even the very files that you are installing and nothing you can do about that.  We are all at WD mercy. 😢 

There is a small chance that by signing the msi you will convince WD but I am not sure. Y(ou will need to try.

And of course, I encourage you to have a look at the repo/codebase to ensure that you are comfortable with what WixSharp is producing.

**Reply by vilu988** [reply]
Thank you for the response. I will try with signing and let you know. 
One more information: I have no custom actions defined, only attach to event to change the images during installation.

**Reply by oleg-shilo** [reply]
"attach to event" is a custom action. It only used a simpler C#-based declaration for that. However of course it's not a valid reason for marking the msi as "compromised".

If you do find the workaround, please share it here. It can be super useful for others.
